spring-kafka/src/main/java/org/springframework/kafka/retrytopic/EndpointCustomizer.java

```diff
@@ -24,21 +24,25 @@
  * Customizes main, retry and DLT endpoints in the Retry Topic functionality
  * and returns the resulting topic names.
  *
+ * @param <T> the listener endpoint type.
+ *
  * @author Tomaz Fernandes
+ * @author Wang Zhiyang
+ *
  * @since 2.7.2
  *
  * @see EndpointCustomizerFactory
  *
  */
 @FunctionalInterface
-public interface EndpointCustomizer {
+public interface EndpointCustomizer<T extends MethodKafkaListenerEndpoint<?, ?>> {
 
 	/**
 	 * Customize the endpoint and return the topic names generated for this endpoint.
 	 * @param listenerEndpoint The main, retry or DLT endpoint to be customized.
 	 * @return A collection containing the topic names generated for this endpoint.
 	 */
-	Collection<TopicNamesHolder> customizeEndpointAndCollectTopics(MethodKafkaListenerEndpoint<?, ?> listenerEndpoint);
+	Collection<TopicNamesHolder> customizeEndpointAndCollectTopics(T listenerEndpoint);
 
 	class TopicNamesHolder {
 
```

Comment:
Modifying the interface signature, which is part of the public API. Although we are just adding a strong type here and it may not affect any implementations, we may also need to mention this in `whats-new.` - the face that `EndpointCustomizer` is strictly for  `MethodKafkaListenerEndpoint` types.